config/src/main/java/org/axonframework/config/DefaultConfigurer.java

```diff
@@ -353,7 +354,7 @@ protected <T> Optional<T> defaultComponent(Class<T> type, Configuration configur
      */
     protected CommandGateway defaultCommandGateway(Configuration config) {
         return defaultComponent(CommandGateway.class, config)
-                .orElseGet(() -> new DefaultCommandGateway(config.commandBus()));
+                .orElseGet(() -> new DefaultCommandGateway(config.commandBus(), new ClassBasedMessageNameResolver()));
```

Comment:
@smcvb should I add `MessageNameResolver` to `Configuration` and use it as `config.messageNameResolver()`? Or we can leave it for now as the Configuration is a subject for the future changes?